src/backup_domain.rs

```diff
@@ -22,3 +22,143 @@ use crate::pac::BKP;
 pub struct BackupDomain {
     pub(crate) _regs: BKP,
 }
+
+
+macro_rules! write_drx {
+    ($self:ident, $drx:ident, $new:expr) => ({
+        let old: u16 = $self._regs.$drx.read().bits() as u16;
+        if old != $new {
```

Comment:
Why is it required to check whether the value has changed before writing it?

Reply:
Good question, I'm not sure if it's done by the lower level code or not. I did it because I heard about wear of the EEPROM, don't know if this will make a difference or not.

Reply:
Correct me if i'm wrong, but I don't think the bkp_drs are implemented using EEPROM since they lose their data between power cycles, they only keep their values while VBAT has power

Reply:
Yeah, IIRC it's regular RAM which is powered by the battery backup (if necessary). If it was any kind of non-volatile storage that wouldn't work either because you need to erase a block of memory before you could write to it. 😉

Reply:
OK, thanks for the clarification. This is my first embedded project :-) I will remove the read and the comparison.